src/Service/Telemetry/ApplicationInsightsLoggerProvider.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Provides a custom logger implementation that sends telemetry events to Azure Application Insights.
+/// See <see href="https://learn.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider">Custom Logging Provider</see> for more information.
+/// </summary>
+public class ApplicationInsightsLoggerProvider : ILoggerProvider
+{
+    private readonly TelemetryClient _telemetryClient;
+
+    public static readonly string ERROR_CAUGHT_EVENT_NAME = "ErrorCaught";
+
+    public ApplicationInsightsLoggerProvider(TelemetryClient telemetryClient)
+    {
+        _telemetryClient = telemetryClient;
+    }
+
+    public ILogger CreateLogger(string categoryName)
+    {
+        return new ApplicationInsightsLogger(_telemetryClient, categoryName);
+    }
+
+    public void Dispose()
+    {
+        // Nothing to dispose
+    }
+
+    /// <summary>
+    /// This logger implementation is used to track errors caught by the application.
+    /// </summary>
+    private class ApplicationInsightsLogger : ILogger
```

Comment:
There is a lot more built in functionality that the library's included `ApplicationInsightsLogger` provides https://github.com/microsoft/ApplicationInsights-dotnet/blob/2.21.0/LOGGING/src/ILogger/ApplicationInsightsLogger.cs

Based on how you configure AppInsights (`applicationInsightsLoggerOptions.TrackExceptionsAsExceptionTelemetry` == true/false) app insights knows to `TrackTrace()` vs `TrackException()`

After looking through app insights docs, I don't think a custom logger to execute `TrackEvent()` is the right way forward at this time. Right now we just want to upload logs emitted by DAB. 

Per API summary, TrackEvent is used for "User actions and other events. Used to track user behavior or to monitor performance." I think we need to rethink what we consider as events and how we should use them. Then, have a follow up PR that adds additional features. Ideally, we want to reduce rolling our own as much as possible. 
https://learn.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#api-summary

Reply:
https://learn.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#trackevent

Reply:
makes sense, removed the class.